opentelemetry/src/lib.rs

```diff
@@ -55,40 +35,109 @@
 //! # }
 //! ```
 //!
+//! See the [examples](https://github.com/open-telemetry/opentelemetry-rust/tree/main/examples) directory for different integration patterns.
+//!
 //! See the [`trace`] module docs for more information on creating and managing
 //! spans.
 //!
 //! [`Span`]: crate::trace::Span
 //!
-//! # Metrics
-//!
-//!
-//! The [`metrics`] module includes types for recording measurements about a
-//! service at runtime.
-//!
-//! ### Creating instruments and recording measurements
-//!
+//! # Getting Started with Metrics
+//!
+//! The [`metrics`] module provides types for recording measurements about a
+//! service at runtime. Below are the key steps to report measurements using
+//! OpenTelemetry Metrics:
+//!
+//! 1. **Obtain a Meter:** Get a `Meter` from a `MeterProvider`.
+//! 2. **Create Instruments:** Use the `Meter` to create one or more instruments
+//!    (e.g., counters, histograms).
+//! 3. **Record Measurements:** Use the instruments to record measurement values
+//!    along with optional attributes.
+//!
+//! ## How Metrics Work in OpenTelemetry
+//! In OpenTelemetry, raw measurements recorded using instruments are
+//! **aggregated in memory** to form metrics. These aggregated metrics are
+//! periodically exported by the [`opentelemetry_sdk`] at fixed intervals (e.g.,
+//! every 60 seconds) via exporters such as [`opentelemetry-stdout`] or
+//! [`opentelemetry-otlp`]. This reduces reporting overhead while ensuring
+//! up-to-date data. The aggregation strategy and export interval can be
+//! customized in the [`opentelemetry_sdk`] based on your use case.
+//!
+//! ## Choosing the Right Instrument
+//! Selecting the correct instrument is critical for accurately representing
+//! your metrics data:
+//!
+//! - Use **Counters** for values that only increase, such as the number of
+//!   requests served or errors encountered.
+//! - Use **UpDownCounters** for values that can increase or decrease, such as
+//!   the number of active connections, number of items in a queue etc.
+//! - **Gauges:** Use for values that can go up or down and represent the
+//!   current state, such as CPU usage, temperature etc.
+//! - Use **Histograms** for measuring the distribution of a value, such as
+//!   response times or payload sizes.
+//!
+//! ### Observable Instruments
+//!
+//! Counters, UpDownCounters, and Gauges have Observable variants that allow
+//! values to be reported through a callback function. Observable instruments
+//! are ideal when the metric value is managed elsewhere and needs to be
+//! observed by OpenTelemetry instrumentation. The callbacks are automatically
+//! invoked by the OpenTelemetry SDK before every export (e.g., every 60
```

Comment:
```suggestion
//! invoked by the OpenTelemetry SDK before every export, force_flush and shutdown.
```

Reply:
Good point. i'll keep iterating to make it more thorough, in follow up PRs.